MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc

```diff
@@ -0,0 +1,127 @@
+      out[index][0] = 0;
+      out[index][1] = 0;
+      out[index][2] = magneticFieldAtPoint(in[index]);
+    }
+  }
+};
+
+int main() {
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  int count = 0;
+  float px, py, pz;
+  float bx, by, bz;
+  vector<Vector3f> points;
+  vector<GlobalVector> referenceB_vec;
+
+  int numberOfPoints = 100;
+  do {
+    if (!(file.read((char*)&px, sizeof(float)) && file.read((char*)&py, sizeof(float)) &&
+          file.read((char*)&pz, sizeof(float)) && file.read((char*)&bx, sizeof(float)) &&
+          file.read((char*)&by, sizeof(float)) && file.read((char*)&bz, sizeof(float))))
+      break;
+
+    const auto gp = GlobalPoint(px, py, pz);
+    if (gp.perp2() > Parameters::max_radius2 || fabs(gp.z()) > Parameters::max_z)
+      continue;
+
+    points.push_back(Vector3f(px, py, pz));
+    referenceB_vec.push_back(GlobalVector(bx, by, bz));
+    count++;
+  } while (count < numberOfPoints);
+
+  const size_t size = points.size();
+  // allocate the input and output host buffer in pinned memory accessible by the Platform devices
+  auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
+  auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
+  // fill the input buffers, and the output buffer with zeros
+  for (size_t i = 0; i < size; ++i) {
+    in_host[i] = points[i];
+    out_host[i] = {0.};
+  }
+
+  float resolution = 0.2;
+  float maxdelta = 0.;
+  int fail = 0;
+
+  // run the test on each device
+  for (auto const& device : cms::alpakatools::devices<Platform>()) {
+    auto queue = Queue(device);
+    // allocate input and output buffers on the device
+    auto points_dev = cms::alpakatools::make_device_buffer<float[]>(queue, size);
+    auto deviceB = cms::alpakatools::make_device_buffer<float[]>(queue, size);
+
+    // copy the input data to the device; the size is known from the buffer objects
+    alpaka::memcpy(queue, points_dev, in_host);
+    // fill the output buffer with zeros; the size is known from the buffer objects
+    alpaka::memset(queue, deviceB, 0.);
+
+    auto workDiv = cms::alpakatools::make_workdiv<Acc1D>(1, size);
+    alpaka::exec<Acc1D>(queue, workDiv, MagneticFieldKernel, points_dev.data(), deviceB.data(), size);
+
+    // copy the results from the device to the host
+    alpaka::memcpy(queue, out_host, deviceB);
+
+    // wait for the kernel and the potential copy to complete
+    alpaka::wait(queue);
+
+    // check the results
+    for (uint i=0; i<points.size(); i++) {
+      const auto point = points[i];
+      const auto referenceB = referenceB_vec[i];
+      const auto out_host_element = out_host[i];
+      GlobalVector parametricB(out_host_element[0], out_host_element[1], out_host_element[2]);
+      if ((referenceB - parametricB).mag() > resolution) {
```

Comment:
Isn't this the same as `delta` two lines below ?
Rather than computing it twice, you can move the declaration of `delta` above.

Reply:
OK: https://github.com/cms-sw/cmssw/pull/44360/commits/8d93f61cc051bd4b2a39d6559170dd00f6a2bcef